src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -0,0 +1,232 @@
+            "TERM",
+            "USER",
+            "XFILESEARCHPATH",
+
+            // Windows system variables
+            "ALLUSERSPROFILE",
+            "APPDATA",
+            "CD",
+            "ClientName",
+            "CMDEXTVERSION",
+            "CMDCMDLINE",
+            "CommonProgramFiles",
+            "COMPUTERNAME",
+            "COMSPEC",
+            "DATE",
+            "ERRORLEVEL",
+            "HighestNumaNodeNumber",
+            "HOMEDRIVE",
+            "HOMEPATH",
+            "LOCALAPPDATA",
+            "LOGONSERVER",
+            "NUMBER_OF_PROCESSORS",
+            "OS",
+            "PATHEXT",
+            "PROCESSOR_ARCHITECTURE",
+            "PROCESSOR_ARCHITEW6432",
+            "PROCESSOR_IDENTIFIER",
+            "PROCESSOR_LEVEL",
+            "PROCESSOR_REVISION",
+            "ProgramW6432",
+            "ProgramData",
+            "ProgramFiles",
+            "ProgramFiles (x86)",
+            "PROMPT",
+            "PSModulePath",
+            "Public",
+            "RANDOM",
+            "%SessionName%",
+            "SYSTEMDRIVE",
+            "SYSTEMROOT",
+            "TEMP", "TMP",
+            "TIME",
+            "UserDnsDomain",
+            "USERDOMAIN",
+            "USERDOMAIN_roamingprofile",
+//            "USERNAME",  // Not whitelisted because this is a likely variable name for credentials binding
+            "USERPROFILE",
+            "WINDIR"
+    ));
+
+    /**
+     * Does first-level sanitization of a step's params, returning the Describable arguments map,
+     * as from {@link org.jenkinsci.plugins.workflow.steps.StepDescriptor#defineArguments(Step)}
+     * @param stepArguments Arguments provided when creating the step
+     * @param variables Environment variables to remove
+     * @return Arguments map, with oversized values and values containing bound credentials stripped out.
+     */
+    @Nonnull
+    public static Map<String,Object> sanitizedStepArguments(@Nonnull Map<String, Object> stepArguments, @CheckForNull EnvVars variables) {
+        HashMap<String, Object> output = new HashMap<String, Object>();
+
+        if (variables == null || variables.size() == 0) {
+            // No need to sanitize against environment variables
+            for (Map.Entry<String,Object> entry : stepArguments.entrySet()) {
+                output.put(entry.getKey(), ArgumentsAction.isOversized(entry.getValue()) ? OVERSIZE_VALUE : entry.getValue());
+            }
+            return output;
+        }
+
+        for (Map.Entry<String,?> param : stepArguments.entrySet()) {
+            Object val = param.getValue();
+            if (ArgumentsAction.isOversized(val)) {
+                output.put(param.getKey(), OVERSIZE_VALUE);
+            } else if (val instanceof String && !isStringSafe((String)val, variables, SAFE_ENVIRONMENT_VARIABLES)) {
+                output.put(param.getKey(), MASKED_VALUE);
+            } else {
+                output.put(param.getKey(), val);
```

Comment:
This will not handle string in nested `Describable`s AFAICT. You need to recurse.